xcdat/regridder/regrid2.py

```diff
@@ -368,6 +382,15 @@ def _extract_bounds(bounds: np.ndarray) -> Tuple[np.ndarray, np.ndarray]:
         lower = bounds[:, 1]
         upper = bounds[:, 0]
 
+    # Make sure to convert the bounds to numpy array beforehand.
+    # Otherwise the error `ValueError: cannot convert float NaN to integer`
+    # is raised when calculating cell weights with `_map_longitude` when
+    # calling `np.sin(np.deg2rad(x))`.
+    if isinstance(lower, Array):
+        lower = lower.compute()
+    if isinstance(upper, Array):
+        upper = upper.compute()
+
```

Comment:
If the dataset is opened using `open_mfdataset()`, we need to convert the Dask Array to `np.ndarray`. I used `.compute()`, but `.values` also works.

Reply:
Removing this code because array conversion isn't actually needed here. Changing `.data` to `.values` in `_get_bounds_ensure_dtype()` means these variables are already `np.ndarray`.